File: SiteVisitsNetCore2020Web/ViewModels/VisitSession.cs
using SiteVisitsNetCore2020Web.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SiteVisitsNetCore2020Web.ViewModels
{
    public class VisitSession
    {
        public string Organization { get; set; }
        /**
        public DateTime firstVisit { get; set; }
        public DateTime lastVisit { get; set; }
    **/
        public DateTime SampleVisitDateTime { get; set; }
        public string SampleIpAddress { get; set; }
        public List<VisitSessionBlock> VisitSessionBlocks { get; set; }
    }
}

File: SiteVisitsNetCore2020Web/ClientApp/src/app/viewmodels/VisitSession.ts
import { VisitSessionBlock } from "./VisitSessionBlock";

export class VisitSession {
  public organization: string;
  sampleIpAddress: string;
  sampleVisitDateTime: Date;

  public visitSessionBlocks: Array<VisitSessionBlock>;

  constructor(options: {
    organization: string,
    sampleIpAddress: string,
    sampleVisitDateTime: string,

    visitSessionBlocks: Array<{
      browser: string,
      device: string,
      visits: Array<any>
    }>
  }) {
    this.organization = options.organization;
    this.sampleIpAddress = options.sampleIpAddress;
    this.sampleVisitDateTime = new Date(options.sampleVisitDateTime);
    this.visitSessionBlocks = options.visitSessionBlocks.map(vsb => new VisitSessionBlock(vsb));
  }
}

